services/network/MbedTlsSession.cpp

```diff
@@ -255,6 +260,12 @@ namespace services
                 {
                     SerializeSessionToFlash(session);
                 });
+
+            if (!storage.back().IsDeserialized())
+            {
+                nvm->erase(&persistedSession);
```

Comment:
You are iterating over nvm and erasing from it. This will cause the iterator to become invalid. Instead, you need to have a loop that allows updating the iterator while erasing